source/extensions/filters/http/jwt_authn/extractor.cc

```diff
@@ -211,6 +216,36 @@ std::vector<JwtLocationConstPtr> ExtractorImpl::extract(const Http::HeaderMap& h
   return tokens;
 }
 
+// as specified in RFC-4648 § 5, plus dot (period, 0x2e), of which two are required in the JWT
+constexpr absl::string_view ConstantBase64UrlEncodingCharsPlusDot =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.";
+
+// Returns a token, not a URL: skips non-Base64Url-legal (or dot) characters, collects following
+// Base64Url+dot string until first non-Base64Url char.
+//
+// The input parameters:
+//    "value_str" - the header value string, perhaps "Bearer string....", and
+//    "after" - the offset into that string after which to begin looking for JWT-legal characters
+//
+// For backwards compatibility, if it finds no suitable string, it returns value_str as-is.
+//
+// It is forgiving w.r.t. dots/periods, as the exact syntax will be verified after extraction.
+//
+// See RFC-7519 § 2, RFC-7515 § 2, and RFC-4648 "Base-N Encodings" § 5.
+absl::string_view ExtractorImpl::extractJWT(const absl::string_view value_str,
```

Comment:
nit: no top-level const parameter (ditto below)

Reply:
I'm confused: isn't that what you asked me to do in an earlier [change request on the same line](https://github.com/envoyproxy/envoy/pull/6384#discussion_r269451976)?

_I.e.,_ do you want me to undo the 1st &amp; 3rd changes in [commit 07b2976&vellip;](https://github.com/larrywest/envoy/commit/07b29766383a305f64d7faa1bfe79183cb5e6777)?

Reply:
I just pushed what I understand to be the requested change, in [commit 152e124&vellip;](https://github.com/envoyproxy/envoy/pull/6384/commits/152e1246245350d609c8eae774cc5415ab274772)

Reply:
No, I meant just passing a `absl::string_view` (without const or reference).

Reply:
Ahhhh. I'm not accustomed to working with `string_view`.

I see now, thanks to [Absiel Tip#1](https://abseil.io/tips/1). Nice.

Thanks.  Change submitted ([commit 7f7ff1b&vellip;](https://github.com/envoyproxy/envoy/pull/6384/commits/7f7ff1bf55c404cdd8aa7152147e6a8690ae718f)).